lib/smack/MemorySafetyChecker.cpp

```diff
@@ -0,0 +1,49 @@
+//
+// This file is distributed under the MIT License. See LICENSE for details.
+//
+#include "smack/MemorySafetyChecker.h"
+#include "smack/Naming.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/Support/Debug.h"
+
+namespace smack {
+
+using namespace llvm;
+
+bool MemorySafetyChecker::runOnModule(Module& m) {
+  DataLayout* dataLayout = new DataLayout(&m);
+  Function* memorySafetyFunction = m.getFunction(Naming::MEMORY_SAFETY_FUNCTION);
+  assert(memorySafetyFunction != NULL && "Couldn't find memory safety function");
+  for (auto& F : m) {
+    if (!Naming::isSmackName(F.getName())) {
+      for (inst_iterator I = inst_begin(F), E = inst_end(F); I != E; ++I) {
+        if (LoadInst* li = dyn_cast<LoadInst>(&*I)) {
+          Value* pointer = li->getPointerOperand();
+          Value* size = ConstantInt::get(IntegerType::get(F.getContext(), 64), 
```

Comment:
Good, good. So then does it make sense that invoking `sizeof` in `smack.c` and in `MemorySafetyChecker.cpp` on the same type might not result in the same number? Meaning, `unsigned long` in `smack.c` might be of different size than `unsigned long` in ``MemorySafetyChecker.cpp`.
This can happen since the two files might not get compiled for the same platforms.